apollo-router/src/query_planner/execution.rs

```diff
@@ -354,6 +354,10 @@ impl PlanNode {
                                     .await;
                                 value.deep_merge(v);
                                 errors.extend(err.into_iter());
+                            } else if current_dir.is_empty() {
+                                // If the condition is on the root selection set and it's the only one
+                                // For queries like {get @skip(if: true) {id name}}
+                                value.deep_merge(Value::Object(Default::default()));
```

Comment:
is this because the initial value passed to `execute_recursively` can be `Value::Null`, as set on line 75 `&initial_value.unwrap_or_default(),`?
If that's the case, I think it would be better to do `&initial_value.unwrap_or_else(|| Value::Object(Default::default())),`, that would avoid those additional checks and possibly prevent future issues in other parts of query plan execution, since we should always work on an object

Reply:
No it's not related, it's a specific case that can only happen with `skip` and `include` directive on a root selection_set